src/NuGet.Core/NuGet.Packaging/Signing/SignVerifier.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Common;
+
+namespace NuGet.Packaging.Signing
+{
+    /// <summary>
+    /// Loads trust providers and verifies package signatures.
+    /// </summary>
+    public class SignVerifier
+    {
+        private readonly List<ISignTrustProvider> _trustProviders;
+        private readonly SignVerifierSettings _settings;
+
+        public SignVerifier(IEnumerable<ISignTrustProvider> trustProviders, SignVerifierSettings settings)
```

Comment:
I'm not seeing a consistent pattern in this PR between using a public constructor or a public static factory method (like [here](https://github.com/NuGet/NuGet.Client/pull/1728/files#diff-610889ae48795bbb62f2f93a59904307R23).